Fix `'TensorBox' object has no attribute 'is_input_buffer'`

jansel opened the pull request:
Summary: Fix for https://fb.workplace.com/groups/1075192433118967/permalink/1664491270855744/

Test Plan: Used reproducer from D74262030

Differential Revision: D74270090




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

Skylion007 commented on torch/_inductor/ir.py:
```diff
@@ -7340,6 +7340,11 @@ def create(data):  # type: ignore[no-untyped-def]
         return TensorBox(StorageBox(data))
 
+    def is_input_buffer(self) -> bool:
+        if hasattr(self.data, "is_input_buffer"):
```
Also check is callable maybe?

jansel committed:
Fix `'TensorBox' object has no attribute 'is_input_buffer'` (#152980)

Summary:

Fix for https://fb.workplace.com/groups/1075192433118967/permalink/1664491270855744/

Test Plan: Used reproducer from D74262030

Differential Revision: D74270090
